asf_heat_pump_affordability/notebooks/investigate_effect_of_room_number.py

```diff
@@ -0,0 +1,496 @@
+        )
+
+        ax.hist(
+            archetype_df[feature],
+            density=True,
+            histtype="bar",
+            alpha=0.5,
+            bins=bin_ranges,
+            color=colour,
+            label=archetype,
+        )
+
+        ax.axvline(
+            archetype_df[feature].median(),
+            color=colour,
+            alpha=0.5,
+            label=f"{archetype} (median)",
+        )
+
+        labels.extend([archetype, f"{archetype}_median"])
+
+    ax.legend(labels=labels)
+
+    ax.set_title(f"{feature}")
+
+
+for f in ["NUMBER_HABITABLE_ROOMS", "TOTAL_FLOOR_AREA"]:
+    generate_plot_archetype(f)
+
+# ## Findings
+# There are differences in average size of home between archetypes:
+# - Flats are smallest on average (~50m2) with fewer rooms (2)
+# - Bungalows and semi-detached/terraced houses are larger (~90m2) with more rooms (4 and 5, respectively)
+# - Detached houses are largest on average (~180m2) with the most rooms on average (7)
+
+# ## 3. Compare coefficients of quantile regression models
+#
+# The code below aims to do the following:
+# 1. Create 2 series of quantile regression models. Each series contains one regression model for each percentile 10-90. One series of models will be trained using the 8 different archetypes as the only features (a_models), the other with the 8 archetypes + number of rooms as the features (ar_models).
+# 2. For each model within both series of models: calculate the difference between pre- and post-1950 coefficients for each of the 4 property types (detached house, semi-detached/terraced house, flat, bungalow). E.g. For Quantile 0.1 model: pre-1950 flats coef - post-1950 flats coef, etc. Calculate the confidence intervals for the difference as well. Ultimately, we get: the difference at each quantile between pre- and post-1950 installation cost for each archetype and the confidence interval for this difference.
+# 3. For each property type, compare the difference in pre- and post-1950 installation cost at each quantile between both model groups (a_models vs ar_models).
+#
+# Note: confidence intervals spanning zero indicate there is a chance the difference in coefficients for pre- and post-1950 archetype is zero and therefore that there may be no meaningful difference in cost of installation when splitting the archetype by pre- or post-1950 build year.
+
+archetypes4 = _data.archetype4.unique()
+X_y = _data[["archetype", "NUMBER_HABITABLE_ROOMS", "adjusted_cost"]]
+
+## Regex pattern to extract archetype names from summary table
+pattern = r"\[(.*?)\]"
+
+# ## Series 1: models with archetypes as only features
+
+# +
+a_model_pre_post_coefs = {
+    "quantile": [],
+    "archetype": [],
+    "pre_1950_coef": [],
+    "post_1950_coef": [],
+    "pre_1950_coef_lower": [],
+    "post_1950_coef_lower": [],
+    "pre_1950_coef_upper": [],
+    "post_1950_coef_upper": [],
+}
+
+for quantile in np.arange(0.1, 1, 0.1):
+    quantile = round(quantile, 1)
```

Comment:
I'm pretty sure this is redundant. I don't think it'd actually fix any floating point precision issues that emerge from `numpy.arange` as I presume that the same size float would be returned so it shouldn't be any more able to represent the floating point value that the original attempt. By the same token I can't see that this would do any harm.

Reply:
Yes I wasnt sure either. When I look at the rounded versions of the values they are all rounded to 1dp, but pre-rounding there are a couple with 10+ dp.